pkg/controller/controller_test.go

```diff
@@ -569,21 +580,70 @@ func TestGetSecretReference(t *testing.T) {
 			expectRef:    nil,
 			expectErr:    true,
 		},
+		"simple - PVC name annotations not supported for Provision and Delete": {
+			secretParams: provisionerSecretParams,
+			params: map[string]string{
+				prefixedProvisionerSecretNameKey: "static-${pv.name}-${pvc.namespace}-${pvc.name}-${pvc.annotations['akey']}",
+			},
+			pvc:       &v1.PersistentVolumeClaim{},
+			expectErr: true,
+		},
 		"template - valid": {
 			secretParams: nodePublishSecretParams,
 			params: map[string]string{
-				nodePublishSecretNameKey:      "static-${pv.name}-${pvc.namespace}-${pvc.name}-${pvc.annotations['akey']}",
+				nodePublishSecretNameKey:      "static-${pv.name}-${pvc.namespace}-${pvc.name}",
 				nodePublishSecretNamespaceKey: "static-${pv.name}-${pvc.namespace}",
 			},
 			pvName: "pvname",
 			pvc: &v1.PersistentVolumeClaim{
 				ObjectMeta: metav1.ObjectMeta{
-					Name:        "pvcname",
-					Namespace:   "pvcnamespace",
-					Annotations: map[string]string{"akey": "avalue"},
+					Name:      "pvcname",
+					Namespace: "pvcnamespace",
 				},
 			},
-			expectRef: &v1.SecretReference{Name: "static-pvname-pvcnamespace-pvcname-avalue", Namespace: "static-pvname-pvcnamespace"},
+			expectRef: &v1.SecretReference{Name: "static-pvname-pvcnamespace-pvcname", Namespace: "static-pvname-pvcnamespace"},
+		},
+		"template - valid, with pvc name": {
+			secretParams: provisionerSecretParams,
+			params: map[string]string{
+				provisionerSecretNameKey:      "${pvc.name}",
+				provisionerSecretNamespaceKey: "ns",
```

Comment:
make this ns something different than the PVC ns so we know we didn't take it from the pvc.